pulsar-broker/src/main/java/com/yahoo/pulsar/broker/web/PulsarWebResource.java

```diff
@@ -146,23 +150,27 @@ protected void validateSuperUserAccess() {
      *             if not authorized
      */
     protected void validateAdminAccessOnProperty(String property) {
-        if (config().isAuthenticationEnabled() && config().isAuthorizationEnabled()) {
-            String appId = clientAppId();
+        validateAdminAccessOnProperty(pulsar(), clientAppId(), property);
+    }
+    
+    protected static void validateAdminAccessOnProperty(PulsarService pulsar, String clientAppId, String property) {
+        if (pulsar.getConfiguration().isAuthenticationEnabled() && pulsar.getConfiguration().isAuthorizationEnabled()) {
+            String appId = clientAppId;
             log.debug("check admin access on property: {} - Authenticated: {} -- role: {}", property,
-                    isClientAuthenticated(), appId);
+                    (clientAppId != null), appId);
```

Comment:
I understand why we can't use `isClientAuthenticated()` here but I think the concrete condition should be in only one place. It might be changed.